### ISSUE TITLE
when client disconnect,  it does not clean the rooms

### ISSUE THREAD

**Issue by notedit** [miguelgrinberg/python-socketio#93]
so  manager.rooms get more and more.

**Comment by miguelgrinberg** [comment]
Can you share your application? I do not see this happening with the example app I have in this repository.

**Comment by notedit** [comment]
this line  i print the rooms,  it will get more and more
https://github.com/notedit/hongpa-backend/blob/master/app.py#L57


then i did some modify here:
https://github.com/notedit/hongpa-backend/blob/master/app.py#L34
manually  leave the room, when on_disconnect.

**Comment by notedit** [comment]
am i doing something wrong?   it really do not remove the client.

**Comment by miguelgrinberg** [comment]
I don't know yet. I have not been able to reproduce this with the examples in this repository, but I have a couple more things to try.

**Comment by notedit** [comment]
may be it is because reconnecting.   i will dig more  when i have time.

**Comment by miguelgrinberg** [comment]
Unfortunately I think I've exhausted my ideas to try to reproduce this. Can you share the client application that you are using with this server, so that I'm running the same thing you have there?

**Comment by miguelgrinberg** [comment]
@notedit I have found a way to get connections to be leaked. In this scenario, the event handler raises an exception. When that happens, I found that no proper cleanup is done. The correct procedure in this case is to disconnect the client, but the raised exception prevents that from happening. Any chance your application event handlers sometimes raise exceptions? You would see them in your log or console output.
